### DISCUSSION TITLE
how to desactive text fields refresh of data provider when clicking outside the text box ?

### DISCUSSION THREAD

**Discussion by nyxdata** [Budibase/budibase#5675] [Help]
I've created a simple app that retrieve email from first name/last name/company name.

To do so, I've created a form containing 3 text fields and a data provider containing a button and a table:

    Form 1
        Text field 1: First Name
        Text field 2: Last Name
        Text field 3: Company Name
        Data Provider 1: Linked to rest API that send the 3 text fields to an AWS endpoint
            Validation button 1: action: refresh data provider
            Table 1: containing the response of my API

Each time I change anything in the text boxes and click outside the box (another field, or anywhere), Budibase refreshes Table 1 (and call my API inducing performance and cost issues)

Do you know how to fix it?

Thank you

**Comment by melohagan** [comment]
Hey @nyxdata 

This is because you're using bindings. So when you change the text field values, the Data Provider is told to go and grab the new information with the new field. 

The way I got around this was to have a button beside the text fields called 'Apply'. This button updates state with an array of values (in your case first, last and company names)

Then you bind your Data Provider to the state, not the text fields, thus it'll only update on the click of the apply button. 

Example here: https://github.com/Budibase/budibase/discussions/5551#discussioncomment-2661937
*Scroll down to 'Desgin - Form Inputs'*
